kie-server-parent/kie-server-services/kie-server-services-prometheus/src/main/java/org/kie/server/services/prometheus/PrometheusKieServerEventListener.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.server.services.prometheus;
+
+import io.prometheus.client.Counter;
+import io.prometheus.client.Gauge;
+import org.kie.server.api.model.KieServerInfo;
+import org.kie.server.services.api.KieContainerInstance;
+import org.kie.server.services.api.KieServer;
+import org.kie.server.services.api.KieServerEventListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PrometheusKieServerEventListener implements KieServerEventListener {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PrometheusKieServerEventListener.class);
+
+    protected static final Counter numberOfContainersStarted = Counter.build()
+            .name("kie_server_container_started_total")
+            .help("Kie Server Started Containers")
+            .labelNames("container_id")
+            .register();
+    protected static final Gauge runningContainers = Gauge.build()
+            .name("kie_server_container_running_total")
+            .help("Kie Server Running Containers")
+            .labelNames("container_id")
+            .register();
+    protected static final Gauge kieServerStartTime = Gauge.build()
```

Comment:
This is an odd metric. Do we really need it?

Reply:
yeah, I research a few examples on how to do this, and will be adding a single metric value to track whenever the server came up. I have one graph in grafana that can then show the uptime for the server.

Reply:
Aha I see, cool! So we'll have one unified metric for this regardless of which KIE service is getting monitored, right?

Reply:
yep